receiver/redisreceiver/redis_metrics.go

```diff
@@ -0,0 +1,157 @@
+package redisreceiver
+
+import (
+	"errors"
+
+	metricsProto "github.com/census-instrumentation/opencensus-proto/gen-go/metrics/v1"
+)
+
+type redisMetric struct {
+	key               string
+	units             string
+	desc              string
+	labels            map[string]string
+	labelDescriptions map[string]string
+	metricType        metricType
+}
+
+func getDefaultRedisMetrics() []*redisMetric {
+	return []*redisMetric{
+		uptimeInSeconds(),
+		uptimeInDays(),
+
+		usedCpuSys(),
+		usedCpuSysChildren(),
+		usedCpuSysUserChildren(),
+
+		lruClock(),
+
+		connectedClients(),
+
+		clientRecentMaxInputBuffer(),
+		clientRecentMaxOutputBuffer(),
+
+		blockedClients(),
+
+		expiredKeys(),
+		evictedKeys(),
+
+		rejectedConnections(),
+
+		usedMemory(),
+		usedMemoryRss(),
+		usedMemoryPeak(),
+		usedMemoryLua(),
+
+		memFragmentationRatio(),
+
+		changesSinceLastSave(),
+
+		instantaneousOpsPerSec(),
+
+		rdbBgsaveInProgress(),
+
+		totalConnectionsReceived(),
+		totalCommandsProcessed(),
+
+		totalNetInputBytes(),
+		totalNetOutputBytes(),
+
+		keyspaceHits(),
+		keyspaceMisses(),
+
+		latestForkUsec(),
+
+		connectedSlaves(),
+
+		replBacklogFirstByteOffset(),
+
+		masterReplOffset(),
+	}
+}
+
+type metricType int
+
+const (
+	unspecified = iota
+	gaugeInt
+	gaugeDouble
+	gaugeDistribution
+	cumulativeInt
+	cumulativeDouble
+)
```

Comment:
Is there any reason that we redeclare this instead of using MetricDescriptor_Type type? The code elsewhere assumes they must be equivalent and uses just type-cast for converting between metricType and MetricDescriptor_Type. In that case what do we gain by having this separate declaration?

Reply:
I wanted to keep this independent of protobuf because it was my understanding that there would be a new data model coming soon that was encoding agnostic, so I didn't want to potentially have to change these values in a bunch of places. Happy to use MetricDescriptor_Type directly though if you think that would be better at this time.

Reply:
Thanks, I understand now the intent. However, because we simply use type conversion between metricType and MetricDescriptor_Type they are inherently tightly coupled although it may not appear so when glancing at the code. If the intent is to make redis metric declarations encoding agnostic then it would be better to have conversion function from metricType to MetricDescriptor_Type and avoid using type conversion.
That is likely an overkill though. We are likely fine with relying on internal data types (and we can assume MetricDescriptor_Type is an internal data type - although the assumption is temporary) and just use MetricDescriptor_Type directly. Once we have the new internal data types we can just update the declaration here. When that happens the compiler will tell us if there are any type mismatches elsewhere, but only if we avoid using type conversions.

Anyway, this is a minor nit for now. Eventually I believe we should feel free to use internal data types within our codebase.

Reply:
Sounds good. Replaced with MetricDescriptor_Type.